libfdproto/utils.c

```diff
@@ -67,18 +56,36 @@ DECLARE_FD_DUMP_PROTOTYPE(fd_sa_dump, sSA * sa, int flags)
 
 void fd_sa_sdump_numeric(char * buf /* must be at least sSA_DUMP_STRLEN */, sSA * sa)
 {
-	char addrbuf[INET6_ADDRSTRLEN];
-	char servbuf[32];
-	
+	char addrbuf[INET6_ADDRSTRLEN+2];
 	if (sa) {
-		int rc = getnameinfo(sa, sSAlen( sa ), addrbuf, sizeof(addrbuf), servbuf, sizeof(servbuf), NI_NUMERICHOST | NI_NUMERICSERV);
-		if (rc) {
+        int rc = fd_addr_to_string(sa, addrbuf);
+  		if (rc) {
 			snprintf(buf, sSA_DUMP_STRLEN, "%s", gai_strerror(rc));
 		} else {
-			snprintf(buf, sSA_DUMP_STRLEN, "%s(%s)", addrbuf, servbuf);
+			snprintf(buf, sSA_DUMP_STRLEN, "%s", addrbuf);
 		}
 	} else {
 		snprintf(buf, sSA_DUMP_STRLEN, "(NULL / ANY)");
 	}
 	
 }
+
+int  fd_addr_to_string(struct sockaddr *sa, char *buf) {
+        const char *addrstr;
+        int rc;
+
+		strcpy(buf, "[");
```

Comment:
Square brackets around an IPv4 address are a bit odd, and I'm not sure it's necessary for IPv6 addresses in this case either? I assume these square brackets are also the reason we have the ugly `INET6_ADDRSTRLEN+2`?

Reply:
The square brackets were just to delimit the IP address.  I can take these out and remove the "+2s" everywhere if you'd prefer it.

For testing (other than homestead UTs) I managed to repro the original problem using tweaked DNS and a series of iptables rules on a system.  The fix then addresses the problem and allows the system to work normally.

I haven't run the live tests over this fix, do you think I need to?

Reply:
I don't feel that strongly about the square brackets, I just found it a bit odd and difficult to reason about so I thought I'd ask the question. Happy for you to do whatever you think is best.

I don't think you need to run the live tests (since I think we're only interested in connections to the HSS, which are set up at start of day and not just-in-time). The kind of things I'm more worried about are...
* Homestead tries to connect to two HSSs, one of which is down. Does Homestead successfully connect to the active HSS? Does Homestead not crash? What about if we're doing IPv6?
Have you tested that kind of thing? If not, are you sure we don't need to?

Reply:
As discussed, tested the failover case from one HSS to the other and seen code go through a IPv6 case even though we haven't been able to test in a full IPv6 environment.